types/validate.go

```diff
@@ -1,4 +1,4 @@
-package main
+package types
```

Comment:
I think it's ok to keep this file in the `main` package and to keep it package-private (lowercase). This isn't a function someone outside this repo would need to use, so we shouldn't expose it. Also, this validation isn't really validating the `types` but rather the vulns, so I don't think it makes sense in `types/`